### ISSUE TITLE
batch normalization affects model.eval's prediction

### ISSUE THREAD

**Issue by dnns92** [pytorch/vision#2079]
## 🐛 Bug
I'm not entirely sure if I maybe do not miss something VERY obvious here, feel free to tell me if that is the case, however I think it might be a bug: Batch normalization should only affect the input during training. However, I find with an easy experiment, that this is not the case. Note that dropout is not applied.

## To Reproduce

0. generate Input "input_"
1. initialize model densenet121
2. set model to eval mode and predict using prediction1 = model(input_)
3. set model to training mode
4. predict something (without model update!)
5. set model to eval mode again
6. predict class from the same input prediction2 = model(input_)
7. note that there was no weight update and the prediction1 != prediction2

```python
from torchvision import models
import torch

def set_parameter_requires_grad(model):
    for param in model.parameters():
        param.requires_grad = False

if __name__ == "__main__":
    model = models.densenet121(pretrained=True)
    set_parameter_requires_grad(model)
    input_ = torch.zeros((1,3, 224, 224))
    model.eval()
    eval_value = model(input_)
    model.train()
    another_variable = model(input_)
    model.eval()
    eval_value_2 = model(input_)

    print(eval_value[0,0:3])
    print(eval_value_2[0,0:3])

###### RETURNS######
tensor([-0.3295,  0.2166, -0.6806])
tensor([-0.5839,  0.4981, -0.4104])
```

## Expected behavior

I expected the model to be independent from batch normalization during model.eval(), i.e. prediction1 == prediction2

## Environment

tested on ubuntu 1804 and windows 10 using python 3.7, torchvision 0.5.0 and torch 1.4.0
